service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -273,4 +278,45 @@ public static String uploadHTML(
 
     return String.format("%s/%s/%s", externs.getBucketPublicUrl(), directoryName, name);
   }
+
+  /**
+   * Load the existing HTML file with the given name from the user uploads S3 bucket.
+   *
+   * @param name the name of the HTML file in S3 to be loaded.
+   * @param directoryName the directory of the file in S3 (without leading or trailing '/').
+   * @return LoadTemplateResponse with the html file name, content, and author.
+   * @throws InvalidURLException if the file does not exist.
+   * @throws BadRequestHTMLException if the HTML file cannot be decoded to a string.
+   * @throws SdkClientException if the loading from S3 failed.
+   */
+  public static LoadTemplateResponse loadHTML(String name, String directoryName) {
+    String HTMLPath = directoryName + "/" + name;
+
+    // if the object does not exist, it throws a 403 for metadata?
+    try {
+      Boolean htmlExists = externs.getS3Client().doesObjectExist(externs.getBucketPublic(), HTMLPath);
+    } catch(AmazonServiceException e) {
+      throw new InvalidURLException();
```

Comment:
I looked into this issue a little and it seems you're right about needing credentials here - when I added credentials to the client builder, `doesObjectExist()` worked as expected.

On line 36 in this file, before we call `build()` on the client builder, can we add this method call: `.withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(PropertiesLoader.loadProperty("aws_access_key"), PropertiesLoader.loadProperty("aws_secret_key"))))`
This should provide the client with the proper credentials to check if objects exist and get `doesObjectExist()` working properly.

Reply:
Thanks for this-- it works properly now (in the helper method)